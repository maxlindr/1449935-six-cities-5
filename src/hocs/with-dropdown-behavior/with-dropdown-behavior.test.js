import React from 'react';
import renderer from 'react-test-renderer';
import PropTypes from 'prop-types';
import withDropdownBehavior from './with-dropdown-behavior';

const MockComponent = () => <div>Mock Component</div>;

MockComponent.propTypes = {
  isOpened: PropTypes.bool.isRequired,
  onToggle: PropTypes.func.isRequired,
  onOptionClick: PropTypes.func.isRequired,
};

const MockComponentWrapped = withDropdownBehavior(MockComponent);

test(`withDropdownBehavior должен корректно отображаться`, () => {
  const tree = renderer.create((
    <MockComponentWrapped
      onChange={() => {}}
      activeOption={`Option`}
      options={{}}
    />
  )).toJSON();

  expect(tree).toMatchSnapshot();
});
