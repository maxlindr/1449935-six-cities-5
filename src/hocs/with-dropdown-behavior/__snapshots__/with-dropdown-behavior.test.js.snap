// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withDropdownBehavior должен корректно отображаться 1`] = `
<div>
  Mock Component
</div>
`;
